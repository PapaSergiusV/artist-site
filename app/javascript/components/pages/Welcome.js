import React from "react"
import PropTypes from "prop-types"
class Welcome extends React.Component {
  render () {
    return (
      <React.Fragment>
        HW
      </React.Fragment>
    );
  }
}

export default Welcome
