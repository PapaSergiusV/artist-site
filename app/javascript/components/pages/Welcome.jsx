import React, { Component, Fragment } from "react";

import FirstScreen from "../shared/FirstScreen";

class Welcome extends Component {
  render () {
    return (
      <Fragment>
        <FirstScreen />
      </Fragment>
    );
  }
}

export default Welcome;
