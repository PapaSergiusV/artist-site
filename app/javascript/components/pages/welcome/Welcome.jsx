import React, { useState } from "react";

import FirstScreen from "../../shared/firstScreen/FirstScreen";
import MainMenu from "../../shared/menu/MainMenu";

const Welcome = props => {
  const [menuIsVisible, toggleMenu] = useState(false);
  return (
    <div>
      <FirstScreen openMenu={() => toggleMenu(true)} />
      {menuIsVisible && <MainMenu {...props} close={() => toggleMenu(false)} />}
    </div>
  );
};

export default Welcome;
