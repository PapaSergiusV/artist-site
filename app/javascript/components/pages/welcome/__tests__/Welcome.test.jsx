import React from "react";
// import { create, update } from "react-test-renderer";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import "../../../../setupTests";

import Welcome from "../Welcome";

describe('Welcome', () => {
  it('should render FirstScreen component only', () => {
    const output = shallow(<Welcome/>);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});