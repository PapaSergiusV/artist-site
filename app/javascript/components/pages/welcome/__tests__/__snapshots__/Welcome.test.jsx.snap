// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render FirstScreen component only 1`] = `
<div>
  <FirstScreen
    openMenu={[Function]}
  />
</div>
`;
